pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go

```diff
@@ -276,27 +312,56 @@ func (ci *CWSInstrumentation) injectForCommand(request *admission.MutateRequest)
 	return mutatePodExecOptions(request.Raw, request.Name, request.Namespace, ci.webhookForCommands.Name(), request.UserInfo, ci.injectCWSCommandInstrumentation, request.DynamicClient, request.APIClient)
 }
 
+func (ci *CWSInstrumentation) resolveNodeArch(nodeName string, apiClient kubernetes.Interface) (string, error) {
+	var arch string
+	// try with the wmeta
+	out, err := ci.wmeta.GetKubernetesNode(nodeName)
+	if err == nil && out != nil {
+		arch = out.Labels["kubernetes.io/arch"]
+	}
+
+	if len(arch) == 0 {
+		// try by querying the api directly
+		node, err := apiClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
```

Comment:
I'd use an informer here instead

Reply:
The previous call is already powered by an informer, yet nothing guarantees it's up to date.

Reply:
Added your proposal @vboulineau 👍🏻

Reply:
Shouldn't we try this only over ifNotFound error then?